relay-event-schema/src/protocol/span_v2.rs

```diff
@@ -0,0 +1,388 @@
+
+    /// Arbitrary attributes on a span.
+    #[metastructure(pii = "true", trim = false)]
+    pub attributes: Annotated<Object<Attribute>>,
+
+    /// Additional arbitrary fields for forwards compatibility.
+    #[metastructure(additional_properties, retain = true, pii = "maybe")]
+    pub other: Object<Value>,
+}
+
+impl SpanV2 {
+    pub fn attribute(&self, key: &str) -> Option<&Annotated<Value>> {
+        Some(&self.attributes.value()?.get(key)?.value()?.value.value)
+    }
+}
+
+/// Status of a V2 span.
+///
+/// This is a subset of OTEL's statuses (unset, ok, error).
+#[derive(Clone, Copy, Debug, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+#[repr(i32)]
+pub enum SpanV2Status {
+    Unset = OtelSpanStatus::Unset as i32,
+    Ok = OtelSpanStatus::Ok as i32,
+    Error = OtelSpanStatus::Error as i32,
+}
+
+impl SpanV2Status {
+    /// Returns the string representation of the status.
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Self::Unset => "unset",
+            Self::Ok => "ok",
+            Self::Error => "error",
+        }
+    }
+}
+
+impl Empty for SpanV2Status {
+    #[inline]
+    fn is_empty(&self) -> bool {
+        false
+    }
+}
+
+impl AsRef<str> for SpanV2Status {
+    fn as_ref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl fmt::Display for SpanV2Status {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+/// Error parsing a `SpanV2Status`.
+#[derive(Clone, Copy, Debug)]
+pub struct ParseSpanV2StatusError;
+
+impl fmt::Display for ParseSpanV2StatusError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "invalid span status")
+    }
+}
+
+impl std::error::Error for ParseSpanV2StatusError {}
+
+impl FromStr for SpanV2Status {
+    type Err = ParseSpanV2StatusError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "unset" => Ok(Self::Unset),
+            "ok" => Ok(Self::Ok),
+            "error" => Ok(Self::Error),
+            _ => Err(ParseSpanV2StatusError),
+        }
+    }
+}
+
+impl FromValue for SpanV2Status {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self>
+    where
+        Self: Sized,
+    {
+        match value {
+            Annotated(Some(Value::String(s)), meta) => Annotated(Self::from_str(&s).ok(), meta),
+            Annotated(_, meta) => Annotated(None, meta),
+        }
+    }
+}
+
+impl IntoValue for SpanV2Status {
+    fn into_value(self) -> Value
+    where
+        Self: Sized,
+    {
+        Value::String(self.to_string())
+    }
+
+    fn serialize_payload<S>(
+        &self,
+        s: S,
+        _behavior: relay_protocol::SkipSerialization,
+    ) -> Result<S::Ok, S::Error>
+    where
+        Self: Sized,
+        S: serde::Serializer,
+    {
+        s.serialize_str(self.as_str())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, ProcessValue)]
+#[repr(i32)]
+pub enum SpanV2Kind {
```

Comment:
This is the same type as `span::SpanKind`, copied here because I couldn't decide where to extract it to. Should probably be deduplicated.

Reply:
We might see the need arise to extend this in the future. As long as we can do it, outside of `OtelSpanKind`, we're good.